cli/main.rs

```diff
@@ -423,7 +423,18 @@ async fn compile_command(
 
   let graph =
     create_graph_and_maybe_check(module_specifier.clone(), &ps, debug).await?;
-  let (bundle_str, _) = bundle_module_graph(graph.as_ref(), &ps, &flags)?;
+
+  let source = if graph.as_ref().modules().len() == 1 {
+    let root_module = graph.as_ref().modules()[0];
+    match root_module.media_type {
+      MediaType::Jsx | MediaType::TypeScript | MediaType::Tsx => {
+        bundle_module_graph(graph.as_ref(), &ps, &flags)?.0
+      }
+      _ => root_module.source.to_string(),
+    }
+  } else {
+    bundle_module_graph(graph.as_ref(), &ps, &flags)?.0
+  };
```

Comment:
You could do something like this (although format this correctly):

```suggestion
  let source = (graph.as_ref().modules().len() == 1).then(|| {
    let root_module = graph.as_ref().modules()[0];
      match root_module.media_type {
        MediaType::JavaScript => Some(root_module.source.to_string())
        _ => None,
      }
  }).flatten().unwrap_or_else(|| bundle_module_graph(graph.as_ref(), &ps, &flags)?.0);  
```